source/Octopus.Tentacle/Scripts/ScriptWorkspaceFactory.cs

```diff
@@ -95,9 +97,56 @@ string FindWorkingDirectory(ScriptTicket ticket)
             return work;
         }
 
+        IScriptWorkspace CreateWorkspace(ScriptTicket scriptTicket, string workingDirectory)
+        {
+            if (!PlatformDetection.IsRunningOnWindows)
+            {
+                return new BashScriptWorkspace(scriptTicket, workingDirectory, fileSystem, sensitiveValueMasker);
+            }
+
+            return new ScriptWorkspace(scriptTicket, workingDirectory, fileSystem, sensitiveValueMasker);
+        }
+
         public string GetWorkingDirectoryPath(ScriptTicket ticket)
         {
-            return fileSystem.GetFullPath(Path.Combine(home.HomeDirectory ?? "", "Work", ticket.TaskId));
+            var baseWorkingDirectory = GetBaseWorkingDirectory();
+            return fileSystem.GetFullPath(Path.Combine(baseWorkingDirectory, ticket.TaskId));
+        }
+        
+        string GetBaseWorkingDirectory()
+        {
+            return Path.Combine(home.HomeDirectory ?? "", WorkDirectory);
+        }
+
+        public List<IScriptWorkspace> GetUncompletedWorkspaces()
```

Comment:
Am I right in thinking these are not necessarily Uncompleted, it's just all the current Workspace Directories.

Reply:
Or Completed = Deleted so Uncompleted = undelete!!

I think the name may be a little awkward!! Or tripped me up anyway!!

Reply:
> Or Completed = Deleted so Uncompleted = not yet deleted!!

Yeah, that was my thinking. Like you say, when we complete, then the directory is deleted.

I thought about "in progress", but that's also not true, as abandoned requests are no longer in progress.

That's why I went for "Uncompleted", as this is true for in progress, as well as abandoned.

🤔 I can't really think of a better name. We could use "Current" if you'd like. Or simply drop any descriptor (e.g. `GetWorkspaces`).